src/adapters/MorphoMarketV1Adapter.sol

```diff
@@ -64,54 +65,54 @@ contract MorphoMarketV1Adapter is IMorphoMarketV1Adapter {
     }
 
     /// @dev Does not log anything because the ids (logged in the parent vault) are enough.
-    /// @dev Returns the ids of the allocation and the potential loss.
-    function allocate(bytes memory data, uint256 assets, bytes4, address)
-        external
-        returns (bytes32[] memory, uint256)
-    {
+    /// @dev Returns the ids of the allocation and the change in allocation.
+    function allocate(bytes memory data, uint256 assets, bytes4, address) external returns (bytes32[] memory, int256) {
         MarketParams memory marketParams = abi.decode(data, (MarketParams));
         Id marketId = marketParams.id();
         require(msg.sender == parentVault, NotAuthorized());
         require(marketParams.loanToken == asset, LoanAssetMismatch());
 
-        uint256 interest = expectedSupplyAssets(marketParams, shares[marketId]).zeroFloorSub(allocation(marketParams));
+        if (shares[marketId] == 0 && assets > 0) allMarketParams.push(marketParams);
```

Comment:
Here you could introduce duplicates if `allocate` is called multiple times and each time the assets round down to 0 shares

Reply:
I'm not sure that allocating allows you to deallocate all shares btw, you still get the rounding

Reply:
fixed by #668